stream-chat-android/src/main/java/com/getstream/sdk/chat/ChatUX.kt

```diff
@@ -0,0 +1,104 @@
+package com.getstream.sdk.chat
+
+import com.getstream.sdk.chat.navigation.ChatNavigationHandler
+import com.getstream.sdk.chat.navigation.ChatNavigator
+import com.getstream.sdk.chat.navigation.ChatNavigatorImpl
+import com.getstream.sdk.chat.style.ChatFonts
+import com.getstream.sdk.chat.style.ChatFontsImpl
+import com.getstream.sdk.chat.style.ChatStyle
+import com.getstream.sdk.chat.utils.strings.ChatStrings
+import com.getstream.sdk.chat.utils.strings.ChatStringsImpl
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.livedata.ChatDomain
+
+/**
+ * ChatUX handles any configuration for the Chat UI elements. It replaces the older Chat class.
+ *
+ * @param client the low level chat client
+ * @param chatDomain the chat domain interface used for offline storage and state
+ * @param fonts allows you to overwrite fonts
+ * @param strings allows you to customize strings
+ * @param navigationHandler navigation handler for customizing things such as the media browsing experience
+ * @param markdown interface to to customize the markdown parsing behaviour, useful if you want to use more markdown modules
+ * @param urlSigner url signing logic, enables you to add authorization tokens for images, video etc
+ *
+ * @see ChatMarkdown
+ * @see UrlSigner
+ * @see ChatStrings
+ * @see ChatFonts
+ */
```

Comment:
this is the most user facing part of the change (hard to change things that are user/developer facing, easy to change internal things.) so lets get it right